### PR TITLE
feat: warm up reporting service

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/reporting.ts [hunk 1]
```diff
@@ -825,5 +825,64 @@ export class ReportingServ {
...
+      const cachedObjs = await cachedContents.getCachedObjects(projectId, appId, region, dataApiRole);
+
+      //warm up redshift serverless
+      if (dashboardCreateParameters.redshift.newServerless) {
+        const input = {
+          Sqls: [`select * from ${appId}.ods_events limit 1`],
+          WorkgroupName: dashboardCreateParameters.redshift.newServerless?.workgroupName ?? undefined,
+          Database: projectId,
+          WithEvent: false,
+          ClusterIdentifier: dashboardCreateParameters.redshift.provisioned?.clusterIdentifier ?? undefined,
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/reporting.ts:851]: You already know it's Redshift Serverless, `ClusterIdentifier` and `DbUser` are useless.

File: src/control-plane/backend/lambda/api/service/reporting.ts [hunk 2]
```diff
@@ -825,5 +825,64 @@ export class ReportingServ {
...
+        logger.info(`Get statement status: ${resp.Status}`);
+        let count = 0;
+        while (resp.Status != StatusString.FINISHED && resp.Status != StatusString.FAILED && count < 60) {
+          await sleep(500);
+          count++;
+          resp = await cachedObjs.redshiftDataClient!.send(checkParams);
+          logger.info(`Get statement status: ${resp.Status}`);
+        }
+        if (resp.Status == StatusString.FAILED) {
+          logger.error('Warmup redshift serverless with error: '+ resp.Status, JSON.stringify(resp));
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/reporting.ts:871]: using `warn` is better

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/reporting.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/reporting.ts]: fixed

File: src/control-plane/backend/lambda/api/service/reporting.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/reporting.ts]: fixed

**Review by zxkane — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Review by tyyzqmf — APPROVED** [review]

**Pull request merged**